add jira tests

dbasunag opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Tests**
	- Introduced a comprehensive suite of unit tests for Jira utility functions, validating behavior for different command line configurations and Jira ID extraction.
	- Added a new configuration section for `pyutils-jira`, specifying parameters for Jira integration, including URL, token, resolved statuses, and target versions.
	- Enhanced test coverage requirements, increasing the threshold from 40 to 65, and improved type-checking configurations for stricter validation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

redhat-qe-bot1 commented:
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

myakove commented:
@coderabbitai review

myakove commented on tests/jira_utils/test_jira_utils.py:
```diff
@@ -0,0 +1,132 @@
+import pytest
+from unittest.mock import MagicMock, patch
```
please use pytest mock

myakove commented:
I think we should test per function, it will be simpler and better tests.
For example to test `get_jira_ids_from_file_content`:

```
@pytest.mark.parametrize(
    "content_and_expected",
    [
        pytest.param({"content": "pytest.mark.jira(ABC-1111)", "expected": {"ABC-1111"}}, id="pytest_mark_jira"),
        pytest.param({"content": "JIRA ID is jira_id=ABC-1111", "expected": {"ABC-1111"}}, id="jira_id="),
        pytest.param(
            {"content": "JIRA URL is https://example.com/browse/ABC-1111", "expected": {"ABC-1111"}}, id="jira_url="
        ),
        pytest.param(
            {
                "content": "pytest.mark.jira(ABC-1111)\nJIRA ID is jira_id=ABC-1112\nJIRA URL is https://example.com/browse/ABC-1113",
                "expected": {"ABC-1111", "ABC-1112", "ABC-1113"},
            },
            id="multiple_jira",
        ),
        pytest.param(
            {
                "content": "pytest.mark.jira(ABC-1111)\nJIRA ID is jira_id=ABC-1111\nJIRA URL is https://example.com/browse/ABC-1111",
                "expected": {"ABC-1111"},
            },
            id="multiple_jira_same_ids",
        ),
        pytest.param({"content": "No Jiera", "expected": set()}, id="no_jira"),
    ],
)
def test_get_jira_ids_from_file_content(content_and_expected):
    jira_ids = get_jira_ids_from_file_content(
        file_content=content_and_expected["content"], issue_pattern=r"([A-Z]+-[0-9]+)", jira_url="https://example.com"
    )
    assert jira_ids == content_and_expected["expected"]
```

and so on .....

dbasunag commented:
> @pytest.mark.parametrize(
>     "content_and_expected",
>     [
>         pytest.param({"content": "pytest.mark.jira(ABC-1111)", "expected": {"ABC-1111"}}, id="pytest_mark_jira"),
>         pytest.param({"content": "JIRA ID is jira_id=ABC-1111", "expected": {"ABC-1111"}}, id="jira_id="),
>         pytest.param(
>             {"content": "JIRA URL is https://example.com/browse/ABC-1111", "expected": {"ABC-1111"}}, id="jira_url="
>         ),
>         pytest.param(
>             {
>                 "content": "pytest.mark.jira(ABC-1111)\nJIRA ID is jira_id=ABC-1112\nJIRA URL is https://example.com/browse/ABC-1113",
>                 "expected": {"ABC-1111", "ABC-1112", "ABC-1113"},
>             },
>             id="multiple_jira",
>         ),
>         pytest.param(
>             {
>                 "content": "pytest.mark.jira(ABC-1111)\nJIRA ID is jira_id=ABC-1111\nJIRA URL is https://example.com/browse/ABC-1111",
>                 "expected": {"ABC-1111"},
>             },
>             id="multiple_jira_same_ids",
>         ),
>         pytest.param({"content": "No Jiera", "expected": set()}, id="no_jira"),
>     ],
> )
> def test_get_jira_ids_from_file_content(content_and_expected):
>     jira_ids = get_jira_ids_from_file_content(
>         file_content=content_and_expected["content"], issue_pattern=r"([A-Z]+-[0-9]+)", jira_url="https://example.com"
>     )
>     assert jira_ids == content_and_expected["expected"]

Added suggested tests and updated name of the test based on your recommendation.

myakove commented on tests/jira_utils/test_jira_utils.py:
```diff
@@ -0,0 +1,186 @@
+    "jira_id, resolved_status, jira_target_versions, target_version_str, file_name, "
+    "skip_project_ids, expected_jira_error_string, "
+    "test_jira_version",
+    [
```
please set `id` for each test

myakove commented on tests/jira_utils/test_jira_utils.py:
```diff
@@ -0,0 +1,186 @@
+LOGGER = get_logger(name=__name__)
+
+
+def test_process_jira_command_line_config_file_empty_config_token(mocker: MockerFixture) -> None:
```
I would not use mock here, have 2 config files and use them to test both cases.

dbasunag replied:
I can use a config file for the first one, but for the second one unless the file passes valid token, url etc, it will again fail. So it must be mocked to get it return something.

myakove replied:
I still do not understand why you need `mocker` here

dbasunag replied:
I don't. I will remove it.

dbasunag commented:
/verified

myakove reviewed (changes requested):
[tool.coverage.report]
fail_under = 40

should be updated.

dbasunag commented:
/verified

dbasunag commented:
/verified
